### Домашнее задание к уроку 5
Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

<b> 1. Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.</b>

In [1]:
import numpy as np
import pandas as pd

In [2]:
def combinations(n, k):
    return (np.math.factorial(n)/(np.math.factorial(k)*np.math.factorial(n-k)))

In [4]:
sa = 16
m = 80
n = 256
y = 0.95

In [5]:
#пользуясь таблицей Лапласа, находим t
Ф_t = y/2
Ф_t

0.475

In [6]:
t = 1.96

In [7]:
#вычислим слагаемое для определения доверительного интервала
d = t*(sa/(256)**0.5)
d

1.96

In [8]:
# мат ожидание лежит между
print (m - t*(sa/(256)**0.5), '< a <', m + t*(sa/(256)**0.5))


78.04 < a < 81.96


<b> 2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95. </b>

In [9]:
z = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
y = 0.95

In [10]:
# выборка малого объема, поэтому применим распределение Стьюдента
# t (0.05, 9) находим по таблице распределения Стьюдента
t = 2.262


In [11]:
# находим выборочное среднее как среднее арифметическое
m = z.sum()/z.size
m

6.590000000000001

In [12]:
# находим среднееквадратичное отклонение через несмещенную выборочную дисперсию
sa = z.std()
sa

0.42766809560686203

In [13]:
#вычислим слагаемое для определения доверительного интервала
d = t*(sa/(z.size)**0.5)
d

0.30591407087612044

In [14]:
# мат ожидание лежит между
print (m - t*(sa/(256)**0.5), '< a <', m + t*(sa/(256)**0.5))


6.52953842298358 < a < 6.650461577016421


<b>3.Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.</b>

In [15]:
# нулевая гипотеза d = 17, альтернативная гипотеза d > 17
# наблюдаемое значение критерия
d = 17
a = 0.05
n = 100
sa = 17.5
s = 4

U = (sa - d)* (n)**0.5/s
U


1.25

In [17]:
# По таблице функции Лапласа находим критическую точку для односторонней критической
# области (при альтернативнйо гипотизе) по уровню значимости α = 0,05 
Ф = (1-2*a)/2
Ф

0.45

In [18]:
t = 1.645
print('Так как U < t, то то нулевую гипотезу можно принять, значит средний диаметр действительно', d, 'мм')

Так как U < t, то то нулевую гипотезу можно принять, значит средний диаметр действительно 17 мм


<b> 4.Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? </b>

In [32]:
z = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
n = 10
d = 200


In [35]:
s = z.var(ddof=1)**0.5
m = z.sum()/z.size
m

198.5

In [36]:
# наблюдаемая величина

U = (m - d)* (n)**0.5/s
U

-1.0651074037450896

По таблице критических точек распределения Стьюдента находим критическую точку по
уровню значимости α = 0.01 и числу степеней свободы k = 9 , откуда  t ≈ 3.25 кр
Так как |U|= 1.065 < t ≈ 3.25, то нулевую гипотезу о равенстве среднего веса 200 г
можно принять